### PR TITLE
add blog posts, ckeditor and friendlyId

### PR DESCRIPTION
**Pull request by georgestam**
From issue #22: @vemv could you please perform a high level review of my code to see if I can improve something?

I did not implement in the end the use of 2 languages for the same post, as the market is only in Spain for now.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -35,6 +35,10 @@ gem 'gibbon'
 # raygun
 gem 'raygun4ruby'
 
+# add posts
+gem 'friendly_id', '~> 5.1.0'
```
Comment (vemv) [inline Gemfile]: 5.1, just two numbers of precision as we do in kuina

File: Gemfile [hunk 2]
```diff
@@ -75,6 +79,7 @@ end
 #  Admin dashboard gems
 gem 'remotipart', github: 'mshibuya/remotipart'
 gem 'rails_admin', '>= 1.0.0.rc'
+gem 'remotipart', github: 'mshibuya/remotipart'
```
Comment (vemv) [inline Gemfile]: add `ref: xxx` where xxx is the commit sha of the used version

File: app/models/article.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Article < ApplicationRecord
+     
+  extend FriendlyId
+  
+  LANGUAGES = ['es', 'en']
+  
+  validates :title, :description, presence: true, length: { minimum: 10 }
+  validates :locale,  inclusion: { in: LANGUAGES, allow_nil: false }
```
Comment (vemv) [inline app/models/article.rb]: simply use `inclusion: LANGUAGES`. by definition that excludes nil

File: app/models/article.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class Article < ApplicationRecord
+     
+  extend FriendlyId
+  
+  LANGUAGES = ['es', 'en']
+  
+  validates :title, :description, presence: true, length: { minimum: 10 }
```
Comment (vemv) [inline app/models/article.rb]: one `validates` per attribute

File: app/models/user.rb [hunk 5]
```diff
@@ -31,7 +31,7 @@ class User < ApplicationRecord
   after_validation :reverse_geocode
 
   after_create :send_welcome_email
-  after_create :subscribe_to_newsletter
+  # after_create :subscribe_to_newsletter
```
Comment (vemv) [inline app/models/user.rb]: never comment out code, that's what git is for (recovering code etc)

File: app/policies/article_policy.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+class ArticlePolicy < ApplicationPolicy
+  class Scope < Scope
+    def resolve
+      scope
+    end
+    
+    def index?
```
Comment (vemv) [inline app/policies/article_policy.rb]: index? and show? go outside of the Scope definition

File: app/views/shared/_home_navbar.html.erb [hunk 7]
```diff
@@ -12,9 +12,9 @@
     <%= link_to url_for( :locale => 'en') do %>
       <%= image_tag "flags/UK_flag.png", class: "flags" %>
     <% end %>
-    
     <%= link_to t(".sign_in", default: "Login"), new_user_session_path, class: "navbar-wagon-item navbar-wagon-link" %>
     <%= link_to t(".sign_up", default: "Sign up"), new_user_registration_path, class: "navbar-wagon-item navbar-wagon-link" %>
+    <%= link_to t(".blog", default: "BLOG"), articles_path, class: "navbar-wagon-item navbar-wagon-link" %>
```
Comment (vemv) [inline app/views/shared/_home_navbar.html.erb]: i18n defaults are bad, I recommend removing all. in exchange you can config i18n to raise exceptions on not found (see kuina development.rb I think)

File: db/migrate/20170313161311_create_articles.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+class CreateArticles < ActiveRecord::Migration[5.0]
+  def change
+    create_table :articles do |t|
+      t.string :title
```
Comment (vemv) [inline db/migrate/20170313161311_create_articles.rb]: add null: false to these attributes

File: db/migrate/20170313161311_create_articles.rb [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+      t.string :description
+      t.string :locale
```
Comment (vemv) [inline db/migrate/20170313161311_create_articles.rb]: remove locale if unused

File: app/models/user.rb (hunk 5 above)
Comment (georgestam) [inline app/models/user.rb]: Once a user is created the `:subscribe_to_newsletter` subscribers to MailChimp. The problem is that each time that I run `seeds.rb` in development, it fails as the user already exists in MailChimp. Any ideas on how can I make `seeds.rb` works even when Mailchimp return a failure? 

Alternatively, how could I run this method in development only?

File: app/policies/article_policy.rb (hunk 6 above)
Comment (georgestam) [inline app/policies/article_policy.rb]: In what cases would be useful to write code inside of the Scope definition?

File: app/views/shared/_home_navbar.html.erb (hunk 7 above)
Comment (georgestam) [inline app/views/shared/_home_navbar.html.erb]: Great, I have change the code to raise the exception in `development.rb`

File: db/migrate/20170313161311_create_articles.rb (hunk 9 above)
Comment (georgestam) [inline db/migrate/20170313161311_create_articles.rb]: It is likely to be implemented, but it is not too urgent. It is good that all the posts are created now with either `es` or `en`.

File: app/policies/article_policy.rb (hunk 6 above)
Comment (vemv) [inline app/policies/article_policy.rb]: Scope is for the `resolve` mechanism only. Read Pundit doc if interested

File: db/migrate/20170313161311_create_articles.rb (hunk 9 above)
Comment (vemv) [inline db/migrate/20170313161311_create_articles.rb]: Make it `null: false` then

File: Gemfile (hunk 2 above)
Comment (georgestam) [inline Gemfile]: It did not work to add the `ref: 'd9109eb032b938672f231155745549a180ee9032'`. I used the current version `1.2`.

**Comment by georgestam** [comment]
@vemv I have implemented the changes that you requested :)

File: app/policies/article_policy.rb [hunk 10]
```diff
@@ -0,0 +1,13 @@
+class ArticlePolicy < ApplicationPolicy
+  class Scope < Scope
+    
+    def index?
```
Comment (vemv) [inline app/policies/article_policy.rb]: still wrong... index and show must be placed outside Scope. Inside ArticlePolicy but outside Scope

Please read Pundit doc for as a refresher

File: db/seeds.rb [hunk 11]
```diff
@@ -221,5 +221,27 @@
...
+    title: Faker::ChuckNorris.fact,
+    description: Faker::Lorem.paragraphs,
+    locale: 'en'
+    })
+end
+
+url = "http://static.giantbomb.com/uploads/original/9/99864/2419866-nes_console_set.png"
+article = Article.first
+article.photo = url
+article.save
```
Comment (vemv) [inline db/seeds.rb]: (almost) never call `.save` instead of `.save!`, remember that habit

**Comment by georgestam** [comment]
Changes implemented. Many thanks Victor 👍

**Pull request merged**